### DISCUSSION TITLE
How would you implement this. (3d sphere time series data using lat/long)

### DISCUSSION THREAD

**Discussion by JiveyGuy** [pyvista/pyvista#4851] [Q&A]
**Goal**: 
Plot a 3d sphere representation of Earth that shows meteorite strikes over time. 

**Question**:
How would you go about this? I have just learned about pyvista and I may not have an achievable idea of how to implement this. My hope is that you reading this are experienced with this package and can advise me on a more achievable approach. 

**My proposed approach:**
Add a horizontal scroll bar that moves the time series forward and shows meteorite strikes as yellow points on the sphere within a certain time range. I am hoping to scale the size of points by mass size and opacity by how recent the strike was. Maybe this is not possible and a plotter open_movie() has to be used. Or even you will advise that this is the wrong package for implementing this. 

The meteorite data is in this format and is from NASA:
| name                         | id    | nametype | recclass              | mass (g)           | fall  | year | reclat     | reclong     | GeoLocation              |
|------------------------------|-------|----------|-----------------------|--------------------|-------|------|------------|-------------|--------------------------|
| Aachen                       | 1     | Valid    | L5                    | 21                 | Fell  | 1880 | 50.775000  | 6.083330    | (50.775, 6.08333)        |
| Aarhus                       | 2     | Valid    | H6                    | 720                | Fell  | 1951 | 56.183330  | 10.233330   | (56.18333, 10.23333)     |

**Comment by edabor** [comment]
Hi @JiveyGuy !

I guess that you can use [`geovista`](https://github.com/bjlittle/geovista), which is an extension of `pyvista`.
Here is some code inspired from https://github.com/pyvista/pyvista/discussions/4661 that could be a good starting point for you.

```python
import geovista as gv
import pyvista as pv
from geovista.common import to_cartesian
import pandas as pd
import datetime
import numpy as np


def points_to_mesh(points, df: pd.DataFrame):
    m = pv.PolyData(points)
    m.point_data["mass"] = df["mass (g)"].fillna(1.0).astype(np.float32)
    return m.glyph(geom=pv.Sphere(0.1,theta_resolution=8, phi_resolution=8), scale="mass", factor=1.0e-6,orient=False)


df = (
    pd.read_csv("Meteorite_Landings.csv", quotechar='"', thousands=",")
    .dropna()
    .astype({"year": int})
)


df[["lon", "lat"]] = (
    df["GeoLocation"]
    .str.replace("(", "")
    .str.replace(")", "")
    .str.split(",", expand=True)
    .astype(float)
)

# convert coordinate to cartesian
points = to_cartesian(lons=df.lon, lats=df.lat)
mesh = points_to_mesh(points, df)

# plot the mesh
plotter = gv.GeoPlotter()
sargs = {"title": "Magnitude", "shadow": True}
kwargs = dict(
    scalar_bar_args=sargs,
    name="points",
    color="red",
)
plotter.add_mesh(mesh, **kwargs)

def callback(value):
    mask = df["year"] == int(value)
    tmp = df[mask]
    if not tmp.empty:
        points = to_cartesian(tmp.lon, tmp.lat)
        mesh = points_to_mesh(points, tmp)
        plotter.add_mesh(mesh, **kwargs)
    else:
        plotter.remove_actor("points")


min = 1900
max = float(datetime.date.today().strftime("%Y"))
plotter.add_slider_widget(callback, [min, max], interaction_event="always", fmt="%.0f")


plotter.add_base_layer(texture=gv.natural_earth_1(), zlevel=0)
plotter.add_graticule()
plotter.add_coastlines()
plotter.add_axes()
plotter.view_xz(negative=True)
plotter.camera.zoom(1.5)
plotter.show()
```
